### ISSUE TITLE
FB16266943: GridRow should support combining for VoiceOver

### ISSUE THREAD

**Issue by samhenrigold** [feedback-assistant/reports#596]
- **Date:** 2024-01-08
- **Resolution:** Open
- **Area:** SwiftUI
- **OS:** iOS
- **Type:** Incorrect/Unexpected Behavior
- **Keywords:** accessibility, 
- **What build does the issue occur on?** iOS 18.1 Seed 6 (22B5069a)
- **Where does the issue occur?** On device

## Description
GridRow does not support combining child views for VoiceOver accessibility using `.accessibilityElement(children: .combine)` which can create a poor VoiceOver experience.

To reproduce:
1. Create a Grid, add a GridRow with more than one subviews
2. Add .accessibilityElement(children: .combine) to the GridRow
3. Enable VoiceOver and interact with the view

Expected result:
VoiceOver should read all combined children in a GridRow as a single element. For example, "Revenue: $500" should be read as one announcement.

Actual result:
VoiceOver reads each GridRow child separately, requiring multiple swipes to hear related information. For example, user must swipe once to hear "Revenue" and swipe again to hear "$500".

Example:
```swift
struct ContentView: View {
    var body: some View {
        Grid {
            GridRow {
                Image(systemName: "dollarsign")
                    .accessibilityHidden(true)
                Text("Revenue")
                Text("$500")
            }
            .accessibilityElement(children: .combine)
            
            GridRow {
                Image(systemName: "chart.line.uptrend.xyaxis")
                    .accessibilityHidden(true)
                Text("Growth")
                Text("27%")
            }
            .accessibilityElement(children: .combine)
        }
    }
}
```

Related to [FB11441750](https://chriswu.com/posts/swiftui/goldenapple/#:~:text=in%20a%20row.-,FB11441750,-SwiftUI%20GridRow%20items) (SwiftUI GridRow items can’t be combined for VoiceOver support)

## Files

https://github.com/user-attachments/assets/2267986e-093f-4162-a0a6-506866eca129
